### ISSUE TITLE
Type of capture is broken in body of lambda with explicit object parameter

### ISSUE THREAD

**Issue by xiaosa-zhz** [llvm/llvm-project#163582]
https://godbolt.org/z/W9aW4Gvfo
https://godbolt.org/z/Yd9qanx55

Replacing explicit object parameter with normal parameter list can be correctly rejected.

**Comment by shafik** [comment]
This report is not clear.

The two pieces of code all work in recent clangs back to clang-19, we don't backport fixes past the last release.

If you still have an issue please post the code that does not work in the summary and the diagnostics. If this works in clang-trunk and the last release then we will consider it fixed.

**Comment by xiaosa-zhz** [comment]
> This report is not clear.
> 
> The two pieces of code all work in recent clangs back to clang-19, we don't backport fixes past the last release.
> 
> If you still have an issue please post the code that does not work in the summary and the diagnostics. If this works in clang-trunk and the last release then we will consider it fixed.

It is accepts-invalid, and trunk is already included (and also accepts the code incorrectly).
I accidently find that the first version this feature is introduced (clang 18) somehow emits correct diagnostics so all compiler versions since then are included.

**Comment by cor3ntin** [comment]
Reduced to

```cpp

int main() {
    [n = 42]<typename Self>(this Self &&) {
        static_assert((decltype(n){}, false));
    }();
}

```
